lib/palindrome_permutation.rb

```diff
@@ -1,4 +1,23 @@
 
 def palindrome_permutation?(string)
-  raise NotImplementedError, "palindrome_permutation? not implemented"
-end
\ No newline at end of file
+  array_string = string.chars
+  palindrome_chars = Hash.new(0) 
+  array_string.each do |ch|
+    palindrome_chars[ch] += 1
+  end
+  char_counts = palindrome_chars.values
+  puts char_counts
+  if string.length % 2 != 0
+    middle_char = char_counts.find_index {|i|
+      i % 2 == 1
+    }
+    if middle_char == nil
+      return false
+    end
+    char_counts.delete_at(middle_char)
```

Comment:
This works, but if I can suggest, just count the number of letters which appear an odd number of times and it should be less than or equal to 1.